File: coding/leetcode/14-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs or len(strs) == 0: return ""
        for i, c in enumerate(strs[0]):
            for s in strs:
                # print(f"c = {c}, s[i]={s[i]}")
                if i >= len(s) or s[i]!=c: 
                    return strs[0][:i]
        return strs[0]

File: coding/leetcode/14-longest-common-prefix/Solution.java
import java.util.ArrayList; 
import java.util.Collections;


class Solution {
    public String longestCommonPrefix(String[] strs) {
        String result = null;
        return result;
    }

    public static void main(String[] args) {

        for (String s : args) {
            System.out.println(s); 
        }
        ArrayList<String> list = new ArrayList<String>();
        list.add("a");
        list.add("c");
        list.add("b");
        Collections.sort(list);  // Collections.reverseOrder()
        System.out.println("after sort:" + list);
    }
}